packages/gcf-utils/test/server.ts

```diff
@@ -38,140 +38,258 @@ const sandbox = sinon.createSandbox();
+          url: `http://localhost:${TEST_SERVER_PORT}/`,
+          headers: {
+            'x-github-delivery': '123',
+            'x-github-event': 'issues',
+            'content-type': 'application/json',
+          },
+          method: 'POST',
+          body: JSON.stringify(payload),
+        });
+        assert.deepStrictEqual(response.status, 200);
+        sinon.assert.calledOnce(enqueueTask);
+        sinon.assert.notCalled(issueSpy);
       });
-      assert.deepStrictEqual(response.status, 200);
-      sinon.assert.notCalled(enqueueTask);
-      sinon.assert.calledOnce(issueSpy);
     });
 
-    it('should handle payloads', async () => {
-      const payload = require(resolve(fixturesPath, './issue_event'));
-      const response = await gaxios.request({
-        url: `http://localhost:${TEST_SERVER_PORT}/`,
-        headers: {
-          'x-github-delivery': '123',
-          'x-cloudtasks-taskname': 'test-bot',
-          'x-github-event': 'issues',
-          'content-type': 'application/json',
-        },
-        method: 'POST',
-        body: JSON.stringify(payload),
+    describe('with verification', () => {
+      let server: http.Server;
+      const bootstrapper = new GCFBootstrapper({
+        projectId: 'test-project',
+        functionName: 'test-bot',
+        location: 'some-location',
       });
-      assert.deepStrictEqual(response.status, 200);
-      sinon.assert.notCalled(enqueueTask);
-      sinon.assert.calledOnceWithMatch(
-        issueSpy,
-        sinon.match.has(
-          'payload',
-          sinon.match({
-            action: 'opened',
-            issue: {
-              number: 10,
+      const issueSpy = sandbox.stub();
+      const repositoryCronSpy = sandbox.stub();
+      const installationCronSpy = sandbox.stub();
+      const globalCronSpy = sandbox.stub();
+      const pubsubSpy = sandbox.stub();
+      let enqueueTask: sinon.SinonStub = sandbox.stub();
+      before(done => {
+        nock.enableNetConnect(host => {
+          return host.startsWith('localhost:');
+        });
+
+        sandbox.stub(bootstrapper, 'getProbotConfig').resolves({
+          appId: 1234,
+          secret: 'foo',
+          webhookPath: 'bar',
+          privateKey: 'cert',
+        });
+
+        // This replaces the authClient with an auth client that uses an
+        // API Key, this ensures that we will not attempt to lookup application
+        // default credentials:
+        sandbox.replace(
+          bootstrapper.storage.authClient,
+          'request',
+          async (opts: object) => {
+            const auth = new GoogleAuth();
+            const client = await auth.fromAPIKey('abc123');
+            return client.request(opts);
+          }
+        );
+        enqueueTask = sandbox.stub(bootstrapper.cloudTasksClient, 'createTask');
+        sandbox
+          .stub(bootstrapper, 'getAuthenticatedOctokit')
+          .resolves(new Octokit());
+
+        server = bootstrapper
```

Comment:
nit: I wonder if it would be worth pulling this into a helper method, as our `before` is ending up with a log of logic (might be nice to have method names that describe it).